### PR TITLE
Add new annotations for common documentation concerns

### PR DESCRIPTION
**Pull request by MinnDevelopment**
[contributing]: https://github.com/DV8FromTheWorld/JDA/wiki/5%29-Contributing

## Pull Request Etiquette

There are several guidelines you should follow in order for your
  Pull Request to be merged.

- [x] I have checked the PRs for upcoming features/bug fixes.
- [x] I have read the [contributing guidelines][contributing].

> It is sometimes better to include more changes in a single commit. 
  If you find yourself having an overwhelming amount of commits, you
  can **rebase** your branch.

## Description

The updated README explains my intentions, questions and opinions welcome.

### CODE REVIEW

File: src/main/java/net/dv8tion/jda/annotations/ForRemoval.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+package net.dv8tion.jda.annotations;
+
+import java.lang.annotation.*;
+
+/**
+ * Functionality annotated with ForRemoval will no longer be supported
+ * and should not be used anymore in new code.
+ */
+@Documented
+@Retention(RetentionPolicy.RUNTIME)
```
Comment (kantenkugel) [inline src/main/java/net/dv8tion/jda/annotations/ForRemoval.java]: I'm pretty sure we don't need them at Runtime... For most of these, SOURCE would probably be enough, and if not, CLASS. I don't see why those would be needed at runtime for eg reflection

File: src/main/java/net/dv8tion/jda/core/JDABuilder.java [hunk 2]
```diff
@@ -716,6 +718,7 @@ public JDA buildAsync() throws LoginException
      *         Use {@link #build()} and {@link JDA#awaitStatus(Status)} instead
      */
     @Deprecated
+    @ReplaceWith("build().awaitReady()")
```
Comment (kantenkugel) [inline src/main/java/net/dv8tion/jda/core/JDABuilder.java]: Wouldn't the better ReplaceWith suggestion be `build().awaitStatus(Status)` here?

File: src/main/java/net/dv8tion/jda/annotations/ForRemoval.java (hunk 1 above)
Comment (MinnDevelopment) [reply src/main/java/net/dv8tion/jda/annotations/ForRemoval.java]: You could write runtime inspections this way

**Comment by MinnDevelopment** [comment]
I have added a custom javadoc tag for incubating features.

image

File: src/main/java/net/dv8tion/jda/annotations/ForRemoval.java (hunk 1 above)
Comment (Shengaero) [reply src/main/java/net/dv8tion/jda/annotations/ForRemoval.java]: I would agree with @kantenkugel here.
If these annotations are really for "documentation" they do not need to be available at runtime via reflection.

**Comment by Shengaero** [comment]
It might also be worthwhile to have rules for applying these annotations documented in the [structure guide](https://github.com/DV8FromTheWorld/JDA/wiki/6%29-JDA-Structure-Guide) section of the wiki, like how to reference methods to use with `@ReplaceWith`.

**Pull request merged**